src/ui/page/share/dataset/steps/metadataFilling/metadataFilling.tsx

```diff
@@ -0,0 +1,336 @@
+/* eslint-disable max-lines-per-function */
+import { useEffect, useMemo, useCallback, type FC } from 'react'
+import { useTranslation } from 'react-i18next'
+import * as O from 'fp-ts/Option'
+import * as IOE from 'fp-ts/IOEither'
+import { pipe } from 'fp-ts/lib/function'
+import type {
+  FormItem,
+  FormItemType,
+  FormItemValue,
+  initFormPayload
+} from '@/ui/store/slice/shareData/shareData.slice'
+import { useAppStore } from '@/ui/store'
+import type { ResourceError } from '@/shared/error'
+import { ShowFileError } from '@/shared/error'
+import { Field } from '@/ui/component/field/field'
+import type { NotificationType } from '@/ui/component/notification/notification'
+import { useDispatchNotification } from '@/ui/hook/useDispatchNotification'
+import './metadataFilling.scss'
+
+type DatasetFormItem = {
+  id: string
+  type: FormItemType
+  label: string
+  value: FormItemValue
+  render: (id: string) => JSX.Element
```

Comment:
I wouldn't pass the `id` as argument here as it's more appropriate to capture it within the closure of the `render` function. This is to ensure that the `id` remains consistent and to avoid the risk of passing an incorrect or mismatched `id` accidentally.

Reply:
I declared id's constants in the `memo` function so the render function can capture them, tell me if it's correct 😌